From 7447b5e5a0eced6e1d1b38da05c1c60e5e0ecd71 Mon Sep 17 00:00:00 2001
From: Antoine Bernard <antoine.bernard@crans.org>
Date: Tue, 22 Aug 2017 00:50:16 +0200
Subject: [PATCH] Fix 695188 intelligent add of Bug#XXXXXX

We prefix the subject of a mail by Bug#XXXXXX: only if the subject does
not already contain it.
---
 scripts/process | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/scripts/process b/scripts/process
index 3eba1fc..b135566 100755
--- a/scripts/process
+++ b/scripts/process
@@ -731,7 +731,13 @@ if ($ref<0) { # new bug report
 print {$debugfh} "maintainers >".join(' ',@maintaddrs)."<\n";
 
 my $orgsender= defined($header{'sender'}) ? "Original-Sender: $header{'sender'}\n" : '';
-my $newsubject= $subject;  $newsubject =~ s/^$gBug#$ref:*\s*//;
+
+my $newsubject;
+if ($subject =~ /$gBug#$ref:*\s*/) {
+    $newsubject = $subject;
+} else {
+    $newsubject = "$gBug#$ref: $subject";
+}
 
 my $xcchdr= $header{ 'x-debbugs-cc' } || '';
 if ($xcchdr =~ m/\S/) {
@@ -829,7 +835,7 @@ my $common_headers='';
 if ($codeletter eq 'U') { # sent to -submitter
     &htmllog("Message", "sent on", $data->{originator}, "$gBug#$ref.");
     my $enc_msg=<<END;
-Subject: $gBug#$ref: $newsubject
+Subject: $newsubject
 Reply-To: $replyto, $ref-quiet\@$gEmailDomain
 ${orgsender}Resent-To: $data->{originator}
 ${resentcc}${common_headers}Resent-Date: $tdate
@@ -851,7 +857,7 @@ END
              (length($data->{package})? "; Package <code>".html_escape($data->{package})."</code>" : '').
              ".");
     my $enc_msg=<<END;
-Subject: $gBug#$ref: $newsubject
+Subject: $newsubject
 Reply-To: $replyto, $ref\@$gEmailDomain
 Resent-From: $header{'from'}
 ${orgsender}Resent-To: $gSubmitList\@$gListDomain
@@ -885,7 +891,7 @@ END
                  ".");
     }
     my $enc_msg=<<END;
-Subject: $gBug#$ref: $newsubject
+Subject: $newsubject
 Reply-To: $replyto, $ref-$baddressroot\@$gEmailDomain
 Resent-From: $header{'from'}
 ${orgsender}Resent-To: $resentccval
-- 
2.14.1

